src/common/src/error.rs

```diff
@@ -80,7 +79,8 @@ pub struct NotImplemented {
     pub issue: TrackingIssue,
 }
 
-#[derive(Error, Debug)]
+#[derive(Error, Debug, Box)]
+#[thiserror_ext(type = RwError, backtrace)]
```

Comment:
`backtrace` here means that a backtrace will be generated if the inner error does not provide one.

https://docs.rs/thiserror-ext/0.0.9/src/thiserror_ext/backtrace.rs.html#26-41


```
/// Capture backtrace if the error does not already have one.
pub struct MaybeBacktrace(Option<Backtrace>);

impl WithBacktrace for MaybeBacktrace {
    fn capture(inner: &dyn std::error::Error) -> Self {
        let inner = if std::error::request_ref::<Backtrace>(inner).is_none() {
            Some(Backtrace::capture())
        } else {
            None
        };
        Self(inner)
    }

    fn provide<'a>(&'a self, request: &mut std::error::Request<'a>) {
        if let Some(backtrace) = &self.0 {
            request.provide_ref(backtrace);
        }
    }
}
```

Reply:
nit: would it be better to let it be like
```suggestion
#[thiserror_ext(boxed = RwError, backtrace)]
```

since `thiserror_ext` does many other things besides boxing

Reply:
Oh, but we now also have `Arc` 😕